src/pulselistener/pulselistener/listener.py

```diff
@@ -90,6 +92,28 @@ def wait_build():
             except Exception as e:
                 logger.error('Failed to queue task from webhook', error=str(e))
 
+            try:
+                # Run risk analysis when the revision is being reviewed by one
+                # of some specific reviewers.
+                run_risk_analysis = False
```

Comment:
Put the analysis detection in a function, and add a unit test.
It will then be easier to move when rebasing #2193

Reply:
Done in 4bae876. I made the test a bit more complex than I could have, simply because this way we can easily build on top of it to test other parts of pulselistener (phabricator.py and part of HookPhabricator are now at least covered in a "good" case).